pool/pool.go

```diff
@@ -49,26 +52,54 @@ func (pool *Pool) Save() {
 func (pool *Pool) Start() {
 	for {
 		select {
-		case api := <-pool.registerChan:
-			pool.Map[api] = true
-			log.Printf("New API: %s\n", api.GetName())
+		case a := <-pool.registerChan:
+			pool.Map[a] = true
+			log.Printf("New API: %s\n", a.GetName())
 			log.Println("Size of Connection Pool: ", len(pool.Map))
 			pool.Save()
-		case api := <-pool.unregisterChan:
-			delete(pool.Map, api)
-			pool.Save()
-			log.Printf("Deleted API: %s\n", api.GetName())
+		case data := <-pool.unregisterChan:
+			delete(pool.Map, data.API)
+			if data.Save {
+				log.Printf("Saving API: %s\n", data.API.GetName())
+				pool.Save()
+			}
+			log.Printf("Deleted API: %s\n", data.API.GetName())
 			log.Println("Size of Connection Pool: ", len(pool.Map))
 		}
 	}
 }
 
+func (pool *Pool) Add(apiName string, id uuid.UUID, ws types.Socket) (types.API, error) {
+	a, err := api.NewAPI(apiName, ws, pool, id)
+	if err != nil {
+		return nil, err
+	}
+	pool.Register(a)
+
+	go a.Run()
```

Comment:
~The `Add` method should not create an API and run it inside, it should just register it to the pool. I would extract both the api creation, and calling `a.Run()` outside of this function.~ Eh, nvm. Makes sense to not expose the API to the handler.